### DISCUSSION TITLE
Dropdown component on mobile iOS(handle long press)

### DISCUSSION THREAD

**Discussion by Lenur29** [ant-design/ant-design#51486] [Q&A]
Hi everybody. I need some help. I use the Dropdown component. As a trigger I set "contextMenu". It works nice on desktop devices. I checked how it works on Andriod mobile device with long press. And long press works fice on Andriod mobile. But it does not work on iOS. For example this component opens a dropdown on Android with long press. Is it possible to acheive the same for iOS? How to handle it correctly?

```
import React from 'react';
import type { MenuProps } from 'antd';
import { Dropdown, theme } from 'antd';

const items: MenuProps['items'] = [
  {
    label: '1st menu item',
    key: '1',
  },
  {
    label: '2nd menu item',
    key: '2',
  },
  {
    label: '3rd menu item',
    key: '3',
  },
];

const App: React.FC = () => {
  const {
    token: { colorBgLayout, colorTextTertiary },
  } = theme.useToken();

  return (
    <Dropdown menu={{ items }} trigger={['contextMenu']}>
      <div
        style={{
          color: colorTextTertiary,
          background: colorBgLayout,
          height: 200,
          textAlign: 'center',
          lineHeight: '200px',
        }}
      >
        Right Click on here
      </div>
    </Dropdown>
  );
};

export default App;
```

**Comment by fr1g1** [comment]
Same problem. Do you have any update?

**Reply by Lenur29** [reply]
Check this implementation - [https://stackoverflow.com/questions/79144607/handle-long-press-in-order-to-display-dropdown-in-ant-design](https://stackoverflow.com/questions/79144607/handle-long-press-in-order-to-display-dropdown-in-ant-design). I works quite good. If you try to open Telegram Web on your iOS, the long press will work very similar to this implementation. Because in Telegram they also show a custom dropdown when user uses long press.
